### PR TITLE
Access Point Cable loss bug has been updated

### PR DESCRIPTION
**Pull request by md-rafeek**
## Description
59877 - Access Points: Observing irrelevant error message when configuring cable loss

## Type of Change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [x] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [x] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/accesspoint_workflow_manager.py [hunk 1]
```diff
@@ -210,15 +217,17 @@
...
             required: False
           radio_role_assignment:
             description: Role assignment mode for the 2.4GHz radio interface. Accepts "Auto", "Client-serving", or "Monitor". For example, Auto.
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the 2.4GHz radio interface. For example, 75.
+            description: |
+              Cable loss in dB for the 2.4GHz radio interface. Valid values range from 0 to 40.
+              This value should be less than the antenna gain value. For example: 2.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:230]: Can you check this? You may get sanity error if you use :
For example: 2

File: plugins/modules/accesspoint_workflow_manager.py [hunk 2]
```diff
@@ -267,7 +276,9 @@
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the 5GHz radio interface. For example, 80.
+            description: |
+              Cable loss in dB for the 5GHz radio interface. Valid values range from 0 to 40.
+              This value should be less than the antenna gain value. For example: 3.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:281]: Same as above

File: plugins/modules/accesspoint_workflow_manager.py [hunk 3]
```diff
@@ -306,15 +317,17 @@
...
             required: False
           radio_role_assignment:
             description: Role assignment mode for the 6GHz radio interface. Accepts "Auto", "Client-serving", or "Monitor".
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the 6GHz radio interface. For example, 75.
+            description: |
+              Cable loss in dB for the 6GHz radio interface. Valid values range from 0 to 40.
+              This value should be less than the antenna gain value. For example: 10.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:330]: Same as above

File: plugins/modules/accesspoint_workflow_manager.py [hunk 4]
```diff
@@ -371,7 +384,9 @@
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the XOR radio interface. For example, 75.
+            description: |
+              Cable loss in dB for the XOR radio interface. Valid values range from 0 to 40.
+              This value should be less than the antenna gain value. For example: 5.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:389]: Same as above

File: plugins/modules/accesspoint_workflow_manager.py [hunk 5]
```diff
@@ -442,7 +457,9 @@
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the TRI radio interface. For example, 75.
+            description: |
+              Cable loss in dB for the TRI radio interface. Valid values range from 0 to 40.
+              This value should be less than the antenna gain value. For example: 6.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:462]: Same as above

File: plugins/modules/accesspoint_workflow_manager.py [hunk 6]
```diff
@@ -1637,6 +1654,13 @@ def validate_radio_parameters(self, radio_config, radio_series, errormsg):
         if antenna_gain and antenna_gain not in range(0, 41):
             errormsg.append("antenna_gain: Invalid '{0}' in playbook".format(antenna_gain))
 
+        cable_loss = radio_config.get("cable_loss")
+        if cable_loss and cable_loss not in range(0, 41):
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:1658]: cable_loss is common in both if and elif, hence you can optimize it

Also you can specify the value in errormsg..

```
if cable_loss:
    if not 0 <= cable_loss <= 40:
        errormsg.append("cable_loss: Invalid '{0}' in playbook. Must be between 0 and 40.".format(cable_loss))
    elif antenna_gain and cable_loss >= antenna_gain:
        errormsg.append("cable_loss: Invalid '{0}' in playbook. Must be less than antenna_gain: {1}.".format(cable_loss, antenna_gain))
```

File: plugins/modules/accesspoint_workflow_manager.py [hunk 7]
```diff
@@ -442,7 +457,9 @@
             type: str
             required: False
           cable_loss:
-            description: Cable loss in dB for the TRI radio interface. For example, 75.
+            description: |
+              Cable loss in dB for the TRI radio interface. Valid values range from 0 to 40.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:461]:
```
    Cable loss in dB for the TRI radio interface. Valid values are from 0 to 40.
    This value must be less than the antenna gain. For example, 6.
```

File: plugins/modules/accesspoint_workflow_manager.py [hunk 8]
```diff
@@ -2280,7 +2304,9 @@ def compare_radio_config(self, current_radio, want_radio):
                     unmatch_count = unmatch_count + 1
                     self.log("Antenna name unmatched: {0}".format(want_radio[dto_key]), "INFO")
                 elif dto_key == "cable_loss":
-                    temp_dtos[dto_key] = want_radio[dto_key]
+                    actual_gain = int(want_radio.get("antenna_gain", 0)) - int(want_radio[dto_key])
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:2307]: Can we split it for clarity? Also you can log the values if required for debugging.

```
                    cable_loss = int(want_radio[dto_key])
                    antenna_gain = int(want_radio.get("antenna_gain", 0))
                    actual_gain = antenna_gain - cable_loss
                    
                    if current_radio.get(self.keymap["antenna_gain"]) != actual_gain:
                        temp_dtos[dto_key] = cable_loss
                        

```

File: plugins/modules/accesspoint_workflow_manager.py [hunk 9]
```diff
@@ -60,14 +60,21 @@
...
           The MAC address used to identify the device. If the MAC address is known,
           it must be provided and cannot be modified. At least one of the following parameters is required
-          to identify the specific access point: mac_address or management_ip_address.
+          to identify the specific access point: mac_address, hostname or management_ip_address.
         type: str
         required: True
       management_ip_address:
         description: |
           The Management IP Address used to identify the device. If the Management IP Address is known,
           it must be provided and cannot be modified. At least one of the following parameters is required
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:69]: Can we say like below for the clarity?

       To identify the specific access point, at least one of the following parameters is required: 
       - mac_address
       - hostname
       - management_ip_address

Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:69]: Updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 1 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:230]: Updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 2 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:281]: Updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 3 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:330]: updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 4 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:389]: updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 7 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:461]: updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 5 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:462]: updated

File: plugins/modules/accesspoint_workflow_manager.py (hunk 6 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:1659]: Updated as per above

File: plugins/modules/accesspoint_workflow_manager.py (hunk 8 above)
Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:2309]: Updated

**Review by md-rafeek — COMMENTED** [review]
Addressed all 9 comments

File: plugins/modules/accesspoint_workflow_manager.py [hunk 10]
```diff
@@ -58,16 +58,26 @@
     suboptions:
       mac_address:
         description: |
-          The MAC address used to identify the device. If the MAC address is known,
-          it must be provided and cannot be modified. At least one of the following parameters is required
-          to identify the specific access point: mac_address or management_ip_address.
+          To identify the specific access point, at least one of the following parameters is required.
```
Comment (madhansansel) [inline plugins/modules/accesspoint_workflow_manager.py:61]:
```
mac_address:
  description: |
    The MAC address used to identify the device. If provided, it cannot be modified. 
    To identify the specific access point, at least one of the following parameters is required.
    - mac_address
    - management_ip_address
    - hostname
```

Comment (md-rafeek) [reply plugins/modules/accesspoint_workflow_manager.py:62]: Updated the same.

**Review by madhansansel — APPROVED** [review]
